# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xf86-video-ps3/segfault.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff --git a/src/ps3_exa.c b/src/ps3_exa.c
index 6f4672f..3eafaf3 100644
--- a/src/ps3_exa.c
+++ b/src/ps3_exa.c
@@ -691,6 +691,9 @@ static void setup_TCL(ScrnInfoPtr pScrn)
 	NV40_LoadVtxProg(pPS3, &nv40_vp);
 	NV40_LoadFragProg(pPS3, &nv30_fp);
 
+	/* ReneR hack */
+	//NV40EXAHackupA8Shaders ();
+
 	PS3DmaKickoff(pPS3);
 	PS3Sync(pPS3);
 }
diff --git a/src/ps3_exa_TCL.c b/src/ps3_exa_TCL.c
index 2991528..9c87def 100644
--- a/src/ps3_exa_TCL.c
+++ b/src/ps3_exa_TCL.c
@@ -441,9 +441,12 @@ NV40EXAPrepareComposite(int op, PicturePtr psPict,
 	}
 
 	if (pdPict->format == PICT_a8)
-		NV40_LoadFragProg(pPS3, nv40_fp_map_a8[fpid]);
-	else
+		// not implemented, yet -ReneR
+		//NV40_LoadFragProg(pPS3, nv40_fp_map_a8[fpid]);
+		return FALSE;
+	else {
 		NV40_LoadFragProg(pPS3, nv40_fp_map[fpid]);
+	}
 
 	/* Appears to be some kind of cache flush, needed here at least
 	 * sometimes.. funky text rendering otherwise :)
